rest_server/src/models/marketplace_item.js

```diff
@@ -165,11 +161,57 @@ class MarketplaceItem {
             },
           ];
         }
-        const items = await this.orm.findAll({ where: filterStatement });
+        const havings = [];
+        if (tags) {
+          havings.push({
+            where: where(
+              fn('array_agg', col('ItemTags.name')),
+              Op.contains,
+              cast(tags, 'varchar[]'),
+            ),
+          });
+        }
+        if (categories) {
+          havings.push({
+            where: where(
+              fn('array_agg', col('ItemCategories.name')),
+              Op.contains,
+              cast(categories, 'varchar[]'),
+            ),
+          });
+        }
+        const items = await this.orm.findAll({
+          where: filterStatement,
+          having: havings,
+          group: [
+            'MarketplaceItem.id',
+            'ItemTags.id',
+            'ItemTags->ItemTagRelation.createdAt',
+            'ItemTags->ItemTagRelation.updatedAt',
+            'ItemTags->ItemTagRelation.MarketplaceItemId',
+            'ItemTags->ItemTagRelation.ItemTagId',
+            'ItemCategories.id',
+            'ItemCategories->ItemCategoryRelation.createdAt',
+            'ItemCategories->ItemCategoryRelation.updatedAt',
+            'ItemCategories->ItemCategoryRelation.MarketplaceItemId',
+            'ItemCategories->ItemCategoryRelation.ItemCategoryId',
+          ],
+          include: [
+            {
+              attributes: ['id', 'name', 'color'],
+              model: this.models.ItemTag.orm,
+              through: { attributes: [] },
+            },
+            {
+              attributes: ['id', 'name'],
+              model: this.models.ItemCategory.orm,
+              through: { attributes: [] },
+            },
+          ],
+        });
         return items;
       },
     );
-
     return await handler(
```

Comment:
er, why deleting this empty line? I quite like it to make the code not such busy...